app/controllers/AmendsController.scala

```diff
@@ -222,4 +207,52 @@ class AmendsController @Inject() (
       }
   }
 
+  private def fetchFixedProtection2016(nino: String)(implicit hc: HeaderCarrier): Future[Option[ProtectionModel]] =
+
+    fetchProtections(nino).map {
+
+      case Right(transformedReadResponseModel: TransformedReadResponseModel) =>
+        filterActiveFixedProtection2016(transformedReadResponseModel)
+
+      case Left(_) =>
+        logger.warn("Unable to retrieve Protection from GET point")
+        None
+    }
+
+  private def fetchProtections(
+      nino: String
+  )(implicit hc: HeaderCarrier): Future[Either[PlaConnectorError, TransformedReadResponseModel]] =
+    plaConnectorV2.readProtections(nino).map(_.map(TransformedReadResponseModel.from))
+
+  private def filterActiveFixedProtection2016(model: TransformedReadResponseModel): Option[ProtectionModel] = {
+    val fixedProtectionType = model.activeProtection.filter(_.isFixedProtection2016)
+    if (fixedProtectionType.isEmpty) {
+      logger.warn("There is no active Fixed Protection 2016")
+      None
+    } else {
+      fixedProtectionType
+    }
+  }
+
+  private def createFixedAndIndividualProtectionModel(
+      model: AmendResponseModel,
+      nino: String
+  )(implicit request: Request[AnyContent]): Future[Either[Result, AmendResponseModel]] =
+    fetchFixedProtection2016(nino).map { modelOption =>
+      modelOption
+        .map { fixedProtectionModel =>
+          val modelCopy = model.protection.copy(
+            protectionReference = fixedProtectionModel.protectionReference,
+            certificateDate = fixedProtectionModel.certificateDate,
+            protectionType = fixedProtectionModel.protectionType
+          )
+          Right(
+            AmendResponseModel(modelCopy)
+          )
+        }
+        .getOrElse(
+          Left(throw Exceptions.RequiredNotFoundProtectionModelException("amendmentOutcome"))
```

Comment:
If you're going to throw exceptions, using `Either` is redundant. The whole purpose of using `Either` is to avoid exceptions and have explicit channel for errors. Functional Programming encourages to use exceptions only for truly exceptional situations and handle business errors explicitly.

So either use `Either` without throwing exceptions, or throw exceptions without `Either`.